src/engine/app.rs

```diff
@@ -230,10 +235,12 @@ impl<T> ApplicationBuilder<T>
     /// Creates a new ApplicationBuilder with the given initial game state and
     /// display configuration.
     pub fn new(initial_state: T, cfg: DisplayConfig) -> ApplicationBuilder<T> {
+        let pool = Arc::new(ThreadPool::new(num_cpus::get()));
```

Comment:
I'm going to use this pool in the next PR, which will allow a shared thread pool.